pkgs/by-name/gl/glfw3/package.nix

```diff
@@ -1,34 +1,39 @@
 { stdenv, lib, fetchFromGitHub, cmake
 , libGL, libXrandr, libXinerama, libXcursor, libX11, libXi, libXext
-, Carbon, Cocoa, Kernel, OpenGL, fixDarwinDylibNames
+, darwin, fixDarwinDylibNames
 , extra-cmake-modules, wayland
 , wayland-scanner, wayland-protocols, libxkbcommon
+, withMinecraftPatch ? false
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   version = "3.4";
-  pname = "glfw";
+  pname = "glfw${lib.optionalString withMinecraftPatch "-minecraft"}";
 
   src = fetchFromGitHub {
     owner = "glfw";
     repo = "GLFW";
-    rev = version;
+    rev = finalAttrs.version;
```

Comment:
This is incorrect and worse than `rec`. See https://github.com/NixOS/nixpkgs/issues/310373

Reply:
While I do agree that it's incorrect, I don't think this is worse than `rec`, especially, as `version` is the only reason `rec` was used before.

I don't think it's in the scope of this PR to fix this inherent issue with `finalAttrs` as long as an alternative convention isn't widely established in Nixpkgs.

Reply:
> While I do agree that it's incorrect, I don't think this is worse than `rec`, especially, as `version` is the only reason `rec` was used before.

No, it is worse. Before the change, not that `rec` is a akin to copy the string textually, overriding `version` will only change the `version` attribute but not `src`, it will work for people using `3+myownpatch` (I don't think everyone like adding suffix to `pname`). With your change, overriding `version` but not `src` will immediately result in a silent unreliable behavior.

> I don't think it's in the scope of this PR to fix this inherent issue with `finalAttrs`.

If you agree this is off-topic, then do not "conveniently" change it in this PR (or "accidentally" for users who don't like it).

Reply:
I am pretty sure we have different priorities here. From the package maintainer's view, the use of rec makes the derivation source harder to comprehend, as it can be unclear where some variables might be coming from. As a compromise, I could do something like the following instead:

```nix
let
  version = "x.y.z";
in
stdenv.mkDerivation {
  ...
  inherit version;

  src = fetchFromGitHub {
    ...
    inherit version;
  };
  ...
}
```